src/argowrapper/routes/routes.py

```diff
@@ -140,6 +142,33 @@ def wrapper(*args, **kwargs):
     return wrapper
 
 
+def check_payment_details(fn):
+    """
+    Check whether user is non-VA user
+    if user is VA-user, do nothing and proceed
+    if user is non-VA user () billing id tag exists in fence user info)
+    add billing Id to argo metadata and pod metadata
+    remove gen3 username from pod metadata
+    """
+
+    @wraps(fn)
+    def wrapper(*args, **kwargs):
+        request = kwargs["request"]
+        request_body = kwargs["request_body"]
+        token = request.headers.get("Authorization")
+        username = argo_engine_helper.get_username_from_token(token)
+        url = request.base_url._url.rstrip("/") + "/user/user"
+        params = {"username": username}
+        user_info = requests.get(url, params, auth=token)
+
+        if "billing_id" in user_info["tags"]:
+            billing_id = user_info["tags"]["billing_id"]
+            request_body.update({"billing_id": billing_id})
```

Comment:
this strategy leaves the door open for `billing_id` misuse as one can add a `billing_id` to the request before sending it to argo-wrapper and we would not see the difference. Please find a different way to pass the retrieved billing id down to the next layers of the code.